src/lib.rs

```diff
@@ -260,6 +260,12 @@ pub use os::OsRng;
 
 pub use isaac::{IsaacRng, Isaac64Rng};
 pub use chacha::ChaChaRng;
+pub use sample::{
+    // TODO: `sample` name will be deprecated in 1.0, use `sample_reservoir` instead
+    sample_reservoir as sample,
+    sample_reservoir,
+    Sample,
+    SampleRef};
```

Comment:
I'm not keen on re-exporting these names. Also, `Sample` conflicts with one of my (to be) proposed changes!

Note: the code *is* all about sampling from sequences, either `Vec` or slices, or iterators over a more general container, so usage of a `seq` module doesn't seem inappropriate to me.

Reply:
we could rename them `SampleContainer` and `SampleRefContainer`. See my comments below for whether we should even keep the traits or go with a function as you suggested elsewhere.